diff --git a/arch/arm/boot/dts/sun7i-a20-olinuxino-lime2.dts b/arch/arm/boot/dts/sun7i-a20-olinuxino-lime2.dts
index 8077f1716fbc..0dbb18e245fd 100644
--- a/arch/arm/boot/dts/sun7i-a20-olinuxino-lime2.dts
+++ b/arch/arm/boot/dts/sun7i-a20-olinuxino-lime2.dts
@@ -46,6 +46,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/interrupt-controller/irq.h>
+#include <dt-bindings/pwm/pwm.h>
 
 / {
 	model = "Olimex A20-OLinuXino-LIME2";
@@ -55,6 +56,13 @@ aliases {
 		serial0 = &uart0;
 	};
 
+	backlight: backlight {
+		compatible = "pwm-backlight";
+		pwms = <&pwm 0 10000 PWM_POLARITY_INVERTED>;
+		brightness-levels = <0 10 20 30 40 50 60 70 80 90 100>;
+		default-brightness-level = <8>;
+	};
+
 	chosen {
 		stdout-path = "serial0:115200n8";
 	};
@@ -82,6 +90,25 @@ led {
 		};
 	};
 
+	panel: panel {
+		compatible = "olimex,lcd-olinuxino-43-ts";
+		backlight = <&backlight>;
+		enable-gpios = <&pio 7 8 GPIO_ACTIVE_HIGH>; /* PH8 */
+		pinctrl-names = "default";
+		pinctrl-0 = <&lcd0_enable_pin &lcd0_rgb888_pins>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		port {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			panel_input: endpoint {
+				remote-endpoint = <&tcon0_out_panel>;
+			};
+		};
+	};
+
 	reg_axp_ipsout: axp_ipsout {
 		compatible = "regulator-fixed";
 		regulator-name = "axp-ipsout";
@@ -159,7 +186,8 @@ phy1: ethernet-phy@1 {
 &mmc0 {
 	vmmc-supply = <&reg_vcc3v3>;
 	bus-width = <4>;
-	cd-gpios = <&pio 7 1 GPIO_ACTIVE_LOW>; /* PH1 */
+	cd-gpios = <&pio 7 1 GPIO_ACTIVE_HIGH>; /* PH1 */
+	cd-inverted;
 	status = "okay";
 };
 
@@ -182,6 +210,11 @@ &pio {
 	vcc-pf-supply = <&reg_vcc3v3>;
 	vcc-pg-supply = <&reg_ldo4>;
 
+	lcd0_enable_pin: lcd0_enable_pin@0 {
+		pins = "PH8";
+		function = "gpio_out";
+	};
+
 	led_pins_olinuxinolime: led-pins {
 		pins = "PH2";
 		function = "gpio_out";
@@ -189,6 +222,12 @@ led_pins_olinuxinolime: led-pins {
 	};
 };
 
+&pwm {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pwm0_pins_a>;
+	status = "okay";
+};
+
 &reg_ahci_5v {
 	gpio = <&pio 2 3 GPIO_ACTIVE_HIGH>;
 	status = "okay";
@@ -259,6 +298,17 @@ &reg_usb2_vbus {
 	status = "okay";
 };
 
+&rtp {
+	allwinner,ts-attached;
+};
+
+&tcon0_out {
+	tcon0_out_panel: endpoint@0 {
+		reg = <0>;
+		remote-endpoint = <&panel_input>;
+	};
+};
+
 &uart0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&uart0_pb_pins>;
